babel-macro/src/macro.ts

```diff
@@ -24,15 +24,48 @@ interface MacrosBabel {
   types: typeof babelTypes;
 }
 
-function findExportedMethods(ast: babelTypes.File, { types: t }: MacrosBabel): string[] {
+function getFunctionName(e: babelTypes.FunctionDeclaration, t: typeof babelTypes): string | undefined {
+  return t.isIdentifier(e.id) ? e.id.name : undefined;
+}
+
+function verifyExportedMember(ast: babelTypes.File, t: typeof babelTypes, funcName: string): void {
+  const body = ast.program.body;
+  const isFunctionExported = body.some((e) =>
+    t.isExpressionStatement(e) &&
+    t.isAssignmentExpression(e.expression) &&
+    e.expression.operator === '=' &&
+    t.isMemberExpression(e.expression.left) &&
+    t.isIdentifier(e.expression.left.object) &&
+    e.expression.left.object.name === 'exports' &&
+    t.isIdentifier(e.expression.left.property) &&
+    e.expression.left.property.name === funcName &&
```

Comment:
iiuc this only specifically matches `exports.funcName = funcName`.
If so, please add a comment. If not please add a comment with the actual example.
Also, please document the fact it is the current typescript generated pattern.

Reply:
Comment added